### PR TITLE
[test][dmfg] Verify rgw realm/zone deployment using spec file

### PR DESCRIPTION
**Pull request by vinpapnoi**
# Description
This test validates the deployment of rgw realm/zones using the orchestrator with a spec file using the below command:
`ceph rgw realm bootstrap --in-file rgw_realm.yaml`

### Changes
Added test file and suite for the test

### Logs
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-PCDKT2/Deploy_custom_RGW_using_spec_file_0.log

### CODE REVIEW

File: cli/ceph/rgw/realm.py [hunk 1]
```diff
@@ -10,13 +10,18 @@ def __init__(self, nodes, base_cmd):
...
 
-    def bootstrap(self, **kw):
+    def bootstrap(self, input=None, **kw):
         """
         Performs rgw realm bootstrap
         Args:
             kw: Key/value pairs of configuration information to be used in the test.
         """
-        cmd = f"{self.base_cmd} bootstrap {build_cmd_from_args(**kw)}"
+        cmd = f"{self.base_cmd} bootstrap "
```
Comment (adityaramteke) [inline cli/ceph/rgw/realm.py:19]: @vinpapnoi Why do we need to remove build_cmd_from_args(**kw)?

Comment (vinpapnoi) [reply cli/ceph/rgw/realm.py:19]: It is moved below. Needed to include "input" variable in the command.

Comment (adityaramteke) [reply cli/ceph/rgw/realm.py:19]: I think you do not have to add extra argument to this method, you can still use build_cmd_from_args, and pass  --in-file as mentioned in ceph rgw realm bootstrap --help:
 -i INPUT_FILE, --in-file INPUT_FILE
 
 @vamahaja correct me if I'm wrong.

Comment (vinpapnoi) [reply cli/ceph/rgw/realm.py:19]: Thanks for the suggestion @adityaramteke 
I tried running the test with your suggested changes and it worked.

**Review by adityaramteke — APPROVED** [review]
LGTM

File: tests/cephadm/test_deploy_rgw_using_spec.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    specs = config.get("specs")
+    realm_name = specs.get("rgw_realm")
+
+    # Enable MGR module
+    CephAdm(installer).ceph.mgr.module.enable("rgw")
+
+    # Bootstrap rgw using new spec file
+    file = create_yaml_config(installer, specs)
+    conf = {"in-file": file}
```
Comment (pranavprakash20) [inline tests/cephadm/test_deploy_rgw_using_spec.py:20]: Hey @vinpapnoi 
One query, is this scenario similar to https://github.com/red-hat-storage/cephci/blob/master/tests/cephadm/test_verify_zone_group_rgw_with_cephadm.py

or it checks with different conf options? 
Please ignore if the scopes are different.

Comment (vinpapnoi) [reply tests/cephadm/test_deploy_rgw_using_spec.py:20]: @pranavprakash20 This test is specific to deploying RGW with a spec file. Above test uses the CLI args to deploy it using the `--realm-name`, `zonegroup-name`, `zone-name` parameters. Here we are using spec file with the `bootstrap` rgw realm option.

Comment (pranavprakash20) [reply tests/cephadm/test_deploy_rgw_using_spec.py:20]: Noted. Thanks for the clarification @vinpapnoi

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**